platform/security/platform-security-core-api/src/main/java/ddf/security/samlp/SimpleSign.java

```diff
@@ -271,26 +270,25 @@ public void validateSignature(Signature signature, Document doc) throws Signatur
         }
     }
 
-    private void validateSignatureAndSamlKey(Signature signature, SAMLKeyInfo samlKeyInfo) throws SignatureException {
+    private void validateSignatureAndSamlKey(Signature signature, SAMLKeyInfo samlKeyInfo)
+            throws SignatureException {
         SAMLSignatureProfileValidator validator = new SAMLSignatureProfileValidator();
         try {
             validator.validate(signature);
-        } catch (ValidationException e) {
+        } catch (org.opensaml.xmlsec.signature.support.SignatureException e) {
             throw new SignatureException("Error validating the SAMLKey signature", e);
         }
 
-        BasicX509Credential credential = new BasicX509Credential();
+        BasicX509Credential credential = null;
         if (samlKeyInfo.getCerts() != null) {
-            credential.setEntityCertificate(samlKeyInfo.getCerts()[0]);
-        } else if (samlKeyInfo.getPublicKey() != null) {
-            credential.setPublicKey(samlKeyInfo.getPublicKey());
+            credential = new BasicX509Credential(samlKeyInfo.getCerts()[0]);
         } else {
-            throw new SignatureException("Can't get X509Certificate or PublicKey to verify signature.");
+            throw new SignatureException(
+                    "Can't get X509Certificate or PublicKey to verify signature.");
         }
-        SignatureValidator sigValidator = new SignatureValidator(credential);
         try {
```

Comment:
Could move this into the `if` block above, declare the `credential` inside that block as well, and I think this would look a little cleaner.